samples/kernel/metairq_dispatch/src/main.c

```diff
@@ -120,10 +118,11 @@ static void record_latencies(struct msg *m, uint32_t latency)
 		return;
 	}
 
-	k_spinlock_key_t key = k_spin_lock(&lock);
-
-	if (stats.num_mirq >= MAX_EVENTS) {
-		k_spin_unlock(&lock, key);
+	/* It might be a potential race condition in this subroutine.
+	 * We check if the msg seqence is reaching the MAX EVENT first.
+	 * To prevent the coming incorrect changes of the array.
+	 */
```

Comment:
I'd still prefer to see a root cause here, and don't understand why the previous fix with the spinlock wasn't correct.  But this looks safe as I read it.